### PR TITLE
MINOR: Added one test and some clarifying comments on tests with simulated EOS

### PR DESCRIPTION
**Pull request by bbejeck**
Added one test and some comments to clarify how EOS is "enabled" for some of the tests.

Ran all streams tests.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
ping @mjsax and @vvcephei for review

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 1]
```diff
@@ -637,6 +641,33 @@ public void shouldNotThrowTaskMigratedExceptionDuringRestoreForChangelogTopicWhe
         assertThat(callback.restored.size(), equalTo(10));
     }
 
+    @Test
+    public void shouldReinitializeStateStoresWhenNoCheckpointFoundAndEOSEnabled() {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:645]: Should we have a reverse test too, that check that a task is not re-initialized if a checkpoint file is found?

**Review by mjsax — APPROVED** [review]
Thanks for th PR @bbejeck. LGTM.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java (hunk 1 above)
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:645]: ack good point I'll add a test for that.

Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:645]: I'll do a follow-on PR for the additional test

**Comment by bbejeck** [comment]
Merged #7626 into trunk.

**Pull request merged**